include/rocksdb/db.h

```diff
@@ -1167,6 +1167,26 @@ class DB {
     return Status::NotSupported("PromoteL0() is not implemented.");
   }
 
+  // Trace DB operations to a file. Use EndTrace() to stop tracing.
+  virtual Status StartTrace(const TraceOptions& /*options*/,
+                            const std::string& /*trace_filename*/) {
+    return Status::NotSupported("StartTrace() is not implemented.");
+  }
+
+  virtual Status EndTrace(const TraceOptions& /*options*/) {
+    return Status::NotSupported("EndTrace() is not implemented.");
+  }
```

Comment:
Once the APIs are there, they are permanent. I'm OK with in the rest part of the code, we give a special implementation just to writing to a file, but for an API, we hope it is generic enough so that in the future the tracing can happen to other places other than a local file. Is there a way, here a tracer object is accepted and we call trace function from the object?

Reply:
Yes, I super-agree with your statement about the APIs, and due to that I did spend considerable time debating how these new APIs should be created. 

I am also debating if these should *not* be made part of the DB class, but instead should be standalone functions taking a db object as a parameter. That way you could pass in ttl_db, geo_db, blob_db or any other type of stackable db to it.

Reply:
Passing a tracer object itself to the APIs ... is an interesting option. Do you mean to say that the users will override how to trace key/values, and create their own tracing formats? I wasn't planning on doing it. 
I was intending that the DB objects itself will decide how to trace, and not giving users an option. So, may be blob_db could have its own way of tracing.

Reply:
Yes very likely someone wants to log in other ways. Someone may upload it to Kafka so that other hosts can replay it in real-time.